File: src/main/java/org/example/bankramenserver/global/error/ErrorResponse.java
package org.example.bankramenserver.global.error;

import lombok.Builder;
import lombok.Getter;
import org.example.bankramenserver.global.error.exception.ErrorCode;

@Getter
@Builder
public class ErrorResponse {

    private final int status;
    private final String code;
    private final String message;

    public static ErrorResponse of(ErrorCode errorCode) {
        return ErrorResponse.builder()
                .status(errorCode.getStatus().value())
                .code(errorCode.name())
                .message(errorCode.getMessage())
                .build();
    }
}

File: src/main/kotlin/org/example/bankramenserver/global/error/ErrorResponse.kt
package org.example.bankramenserver.global.error

import org.example.bankramenserver.global.error.exception.ErrorCode

data class ErrorResponse(val status: Int, val code: String, val message: String) {
    companion object { @JvmStatic fun of(errorCode: ErrorCode) = ErrorResponse(errorCode.status.value(), errorCode.name, errorCode.message) }
}
